include/switch-language.php

```diff
@@ -0,0 +1,173 @@
+<?php
+/**
+ * @package Polylang
+ */
+
+/**
+ * Class to handle site language switch.
+ */
+class PLL_Switch_Language {
+
+	/**
+	 * @var PLL_Model
+	 */
+	private static $model;
+
+	/**
+	 * The previous language.
+	 *
+	 * @var PLL_Language|null
+	 */
+	public static $previous_language;
+
+	/**
+	 * The original language.
+	 *
+	 * @var PLL_Language|null
+	 */
+	private static $original_language;
+
+	/**
+	 * The current language.
+	 *
+	 * @var PLL_Language|null
+	 */
+	private static $current_language;
+
+	/**
+	 * Setups filters.
+	 *
+	 * @since 3.7
+	 *
+	 * @param PLL_Model $model Instance of `PLL_Model`.
+	 * @return void
+	 */
+	public static function init( PLL_Model $model ) {
+		self::$model = $model;
+
+		add_action( 'pll_language_defined', array( static::class, 'set_current_language' ) );
+	}
+
+	/**
+	 * Sets the current language.
+	 *
+	 * @since 3.7
+	 *
+	 * @param string $slug Current language slug.
+	 * @return void
+	 */
+	public static function set_current_language( $slug ) {
+		$language = self::$model->languages->get( $slug );
+		self::$current_language = ! empty( $language ) ? $language : null;
+	}
+
+	/**
+	 * Switches to the given language.
+	 * Hooked on `pll_post_synchronized` at first.
```

Comment:
[Here](https://github.com/polylang/polylang-pro/pull/2239/files#diff-2c7d1cb4435f5895a8afc187ec0bed19bc144797e4a61cf638fd8c567921e1c3R79). 

It would be weird to add a Polylang Pro action to PLL.